components/eamxx/src/share/io/scream_output_manager.cpp

```diff
@@ -550,6 +550,12 @@ void OutputManager::run(const util::TimeStamp& timestamp)
       if (filespecs.file_needs_flush()) {
         flush_file (filespecs.filename);
       }
+
+      // Check if we have hit the max number of snapshots and need to close the file
+      if (filespecs.file_is_full()) {
```

Comment:
Instead of using this method, we should use

```
  if (filespecs.storage.snapshot_fits (m_output_control.next_write_ts))
```

Caveat: you need to ensure that `m_output_control.compute_next_write_ts()` has been called before you attempt to use `next_write_ts`. I believe at the point where you added these mods, this will be the case, but you may want to double check.

Reply:
yes, just confirmed that `compute_next_write_ts()` is called before this chunk of code.

Reply:
@bartgol do you mean the negation of snapshot_fits?

Reply:
Ah, yes, good call. @AaronDonahue see andrew's comment, so don't just copy paste the line I wrote.